java-connectors/kafka-connect-gcp-common/src/main/java/io/lenses/streamreactor/connect/gcp/common/config/AuthModeSettings.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.gcp.common.config;
+
+import io.lenses.streamreactor.common.config.base.ConfigMap;
+import io.lenses.streamreactor.connect.gcp.common.auth.mode.*;
+import lombok.AllArgsConstructor;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigException;
+
+@AllArgsConstructor
+public class AuthModeSettings {
+
+    private final String connectorPrefix;
+
+    public String getAuthModeKey() {
+        return generateKey("gcp.auth.mode");
+    }
+
+    public String getCredentialsKey() {
+        return generateKey("gcp.credentials");
+    }
+
+    public String getFileKey() {
+        return generateKey("gcp.file");
+    }
+
+    private String generateKey(String suffix) {
+        return String.format("%s.%s", connectorPrefix, suffix);
+    }
+
+
+    public ConfigDef withAuthModeSettings(ConfigDef configDef) {
+        return configDef.define(
+                        getAuthModeKey(),
+                        Type.STRING,
+                        AuthModeEnum.DEFAULT.name(),
+                        Importance.HIGH,
+                        "Authenticate mode, 'credentials', 'file' or 'default'"
+                        )
+                .define(
+                        getCredentialsKey(),
+                        Type.PASSWORD,
+                        "",
+                        Importance.HIGH,
+                        "GCP Credentials if using 'credentials' auth mode."
+                        )
+                .define(
+                        getFileKey(),
+                        Type.STRING,
+                        "",
+                        Importance.HIGH,
+                        "File containing GCP Credentials if using 'file' auth mode"
```

Comment:
MINOR: I'm being picky here but what is the format? `file://path/to/file`? May be worth mentioning.

Reply:
Great point.  We do want this to be easy to use.

`file:/` would be a URL, so it's not that.

What makes it hard to document is that it is OS specific - I have added a unix example.